cpp/es3.1/PhoneBook.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+struct Contact {
+    string nome;
+    string cognome;
+    string numTelefono;
+};
+
+class PhoneBook {
+private:
+    vector<Contact> contatti;
+
+public:
+    void append(const Contact& contatto) {
+        contatti.push_back(contatto);
+    }
+
+    Contact* find(const string& nome) {
+        for (auto& contatto : contatti) {
+            if (contatto.nome == nome) {
+                return &contatto;
+            }
+        }
+        return nullptr;
+    }
+
+    vector<Contact> filter(int scelta) {
+        vector<Contact> rubricaFiltrata;
+
+        switch (scelta) {
+            case 1: 
+                char letteraIniziale;
+                cout << endl << "Inserire la lettera iniziale per cui applicare il filtro: ";
+                cin >> letteraIniziale;
+
+                for (const auto& contatto : contatti) {
+                    if (!contatto.nome.empty() && contatto.nome[0] == letteraIniziale) {
+                        rubricaFiltrata.push_back(contatto);
+                    }
+                }
+                break;
+            
+            case 2: 
+                int numElementi;
+                cout << endl << "Inserisci il numero di elementi per il filtro: ";
+                cin >> numElementi;
+
+                int count = 0;
+                for (const auto& contatto : contatti) {
+                    rubricaFiltrata.push_back(contatto);
+                    count++;
+                    if (count >= numElementi) {
+                        break;
+                    }
+                }
+                break;
+        }
+
+        if(scelta < 1 || scelta > 2) {
+                cout << "Scelta non valida." << endl;
+        }
+
+        return rubricaFiltrata;
+    }
+
+    friend ostream& operator<<(ostream& out, const PhoneBook& phoneBook) {
+        string nomeDefault = "Marco";
```

Comment:
Non ho capito perché ti serva un contatto di default...

Reply:
L'avevo inserito per non lasciare vuota la rubrica in caso non venga inserito nessun contatto, è una cosa puramente estetica, posso rimuoverla.